lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -35,8 +35,19 @@ def serializable_hash_for_single_resource(options)
 
         def resource_relationships(options)
           relationships = {}
+
+          association_fields_hash = {}
+          if options[:fields]
+            association_fields_hash = Hash[
+              *options[:fields].select { |s| s.is_a? Hash }
+            ]
+          end
+
           serializer.associations(@include_tree).each do |association|
-            relationships[association.key] = relationship_value_for(association, options)
+            relationships[association.key] = relationship_value_for(
+              association,
+              options.merge(fields: association_fields_hash[association.name])
```

Comment:
not the case since this runs only for associations. The case you said would only happen with the following (given that there is a user association):
```
render json: @videos.limit(1), each_serializer: VideoSerializer,  fields: [:id, user: []]
```

Plus, test would cry out if that was true.